mutations.py

```diff
@@ -0,0 +1,54 @@
+import random
+import os
+
+
+class BaseMutation:
```

Comment:
It is not clear by the name that this is an interface class. It is customary to use the 'I' prefix (that is, to name this class "IMutation").